src/biogeophys/BalanceCheckMod.F90

```diff
@@ -186,28 +331,38 @@ subroutine BeginWaterBalanceSingle(bounds, &
     logical                   , intent(in)    :: use_aquifer_layer    ! whether an aquifer layer is used in this run
     !
     ! !LOCAL VARIABLES:
-    integer :: c, j, fc                  ! indices
+    integer :: c, fc                  ! indices
     !-----------------------------------------------------------------------
 
     associate(                                               &
-         zi         => col%zi                           , & ! Input:  [real(r8) (:,:) ]  interface level below a "z" level (m)
-         zwt        => soilhydrology_inst%zwt_col       , & ! Input:  [real(r8) (:)   ]  water table depth (m)
-         aquifer_water_baseline => waterstate_inst%aquifer_water_baseline, & ! Input: [real(r8)] baseline value for water in the unconfined aquifer (wa_col) for this bulk / tracer (mm)
-         wa         => waterstate_inst%wa_col           , & ! Output: [real(r8) (:)   ]  water in the unconfined aquifer (mm)
+         zi        => col%zi                            , & ! Input:  [real(r8) (:,:) ]  interface level below a "z" level (m)
+         zwt       => soilhydrology_inst%zwt_col        , & ! Input:  [real(r8) (:)   ]  water table depth (m)
+         aquifer_water_baseline => waterstate_inst%aquifer_water_baseline, &  ! Input: [real(r8)] baseline value for water in the unconfined aquifer (wa_col) for this bulk / tracer (mm)
+         wa        => waterstate_inst%wa_col            , & ! Output: [real(r8) (:)   ]  water in the unconfined aquifer (mm)
+         wa_reset_nonconservation_gain => waterbalance_inst%wa_reset_nonconservation_gain_col                                           , & ! Output: [real(r8) (:)   ]  mass gained from resetting water in the unconfined aquifer, wa_col (negative indicates mass lost) (mm)
```

Comment:
...only a formatting change in the `associate` line above. I will try to remove it in the next commit.